src/TBuffer.cpp

```diff
@@ -2155,6 +2160,79 @@ void TBuffer::decodeOSC(const QString& sequence)
             resetColors();
         }
         break;
+    case static_cast<quint8>('8'): {
+        // Handle OSC 8 hyperlinks in the form: "8;params;URI"
+        // Skip the selector "8" but retain semicolons for parsing
+#if defined(DEBUG_OSC_PROCESSING)
+        qDebug().noquote() << "[OSC 8] Raw sequence: " << sequence;
+        qDebug().noquote() << "[OSC 8] Raw hex: " << sequence.toUtf8().toHex(' ');
+#endif
+        QStringView rest = QStringView(sequence).mid(1);  // skip selector "8;"
+        int firstSemi = rest.indexOf(';');
+
+        if (firstSemi == -1) {
+            qWarning() << "OSC 8: Missing first semicolon";
+            return;
+        }
+
+        int secondSemi = rest.indexOf(';', firstSemi + 1);
+
+        if (secondSemi == -1) {
+            qWarning() << "OSC 8: Missing second semicolon";
+            return;
+        }
+
+        QString param = rest.left(firstSemi).toString();
+        QString url = rest.mid(secondSemi + 1).toString();
+
+        // OSC 8 ;; with empty param and URI closes the hyperlink
+        if ((param.isEmpty() && url.isEmpty())) {
+            mCurrentHyperlinkUrl.clear();
+            mCurrentHyperlinkCommand.clear();
+            mCurrentHyperlinkHint.clear();
+            mCurrentHyperlinkLinkId = 0;
+            mHyperlinkActive = false;
+            break;
+        }
+
+        if (!url.isEmpty()) {
+            // Reject unsupported or overly long URLs
+            if (url.length() > 2048) {
+                qWarning() << "TBuffer::decodeOSC(...) - Rejected hyperlink: URL too long:" << url;
+                return;
+            }
+
+            QUrl qurl(url);
+            const QString scheme = qurl.scheme().toLower();
+
+            QStringList command;
+            QStringList hint;
+
+            if (scheme == "send") {
+                QString sendCommand = qurl.path(); // or url.mid(5);
+                command = { QString("send([[%1]])").arg(sendCommand) };
```

Comment:
Usually we use the `qsl` macro in place of `QString` to get the object pre-created at compile time. This applies to here and other cases down below.